apps/air-discount-scheme/backend/src/app/modules/user/user.model.ts

```diff
@@ -29,10 +26,14 @@ export class AirlineUser implements TAirlineUser {
     this.middleName = user.middleName
     this.lastName = user.lastName
     this.gender = user.gender
-    this.nationalId = user.nationalId
+    this.nationalId = AirlineUser.maskNationalId(user.nationalId)
```

Comment:
The User calls super from its derived class and will also have the masked field.
Needs changes.

Reply:
Nice catch!